File: Strivers-A2Z-DSA-Sheet/06.Bit Manipulation/1. Learn Bit Manipulation/javascript/02.Check for the ith bit.js
/**
QUESTION:
Given a number `n` and zero-based bit index `k`, determine if the kth bit is set.

Approach:
Bitwise-AND `n` with (1 << k). A non-zero result indicates that the bit is set.

Time Complexity: O(1)
Space Complexity: O(1)
*/

function checkKthBit(n, k) {
    return (n & (1 << k)) !== 0;
}

module.exports = { checkKthBit };

File: Strivers-A2Z-DSA-Sheet/06.Bit Manipulation/1. Learn Bit Manipulation/python/02.Check for the ith bit.py
"""QUESTION:
Given an integer `n` and zero-based bit index `k`, return whether the kth bit is set.

Approach:
Bitwise-AND `n` with (1 << k). A non-zero result indicates the bit is set.

Time Complexity: O(1)
Space Complexity: O(1)
"""

def check_kth_bit(n: int, k: int) -> bool:
    return (n & (1 << k)) != 0
